File: tests/Lang.Tests/Samples/BadClient.cs
// Sample: poorly-designed client that fails multiple checks
public class BadClientOptions { }

public class BadClient
{
    public BadClient(string connectionString)
    {
        var x = connectionString;
    }

    public async Task<string> GetItemAsync(string id)
    {
        var result = "hello";
        Thread.Sleep(100);
        return result;
    }

    public void DoSomething()
    {
        try
        {
            Console.WriteLine("working");
        }
        catch (Exception ex)
        {
            throw;
        }
    }

    public void DoSomethingBad()
    {
        try
        {
            Console.WriteLine("working");
        }
        catch (Exception ex)
        {
            // swallowed — no rethrow
        }
    }
}

File: tests/Lang.Tests/Samples/bad_client.py
class BadClient:
    """A poorly-designed Python client."""

    def __init__(self, endpoint):
        self._endpoint = endpoint

    def fetch_data(self, query):
        print("fetching...")
        return {}

    def process(self, data):
        try:
            result = data
        except:
            pass

        try:
            result = data
        except Exception as e:
            pass

        return data

    def _internal(self):
        pass
